### DISCUSSION TITLE
bitwarden_rs bug whit docker-compose and  docker run

### DISCUSSION THREAD

**Discussion by Ran-Xing** [dani-garcia/vaultwarden#1318] [General]
Sorry, I can't understand English
left : docker-compose up -d            right: docker run
![image](https://user-images.githubusercontent.com/73819467/104731271-6d832200-5776-11eb-8ac5-a2ab4bd63bc0.png)

![image](https://user-images.githubusercontent.com/73819467/104731535-cb176e80-5776-11eb-9ecd-2857c79b8ad3.png)

**Comment by BlackDex** [comment]
So, what is the bug then? It seems started just fine except that it couldn't read the ssl cert correctly.
Also, you have duplicate ports there, if you do not add /tcp or /udp it defaults to both, so adding /udp is useless.

Other than that it looks ok, it starts, shows on which ports it is running.

**Reply by Ran-Xing** [reply]
I'll test it today -p 80:80 does not contain udp, you can use ncat test.

docker run --it --rm ubuntu bash
apt update && apt install net-tools ncat -y
ifconfig
ncat -luvp 5555


ncat localhost 5555 -uvz

**Reply by BlackDex** [reply]
http 1.x and 2.x do not work on UDP, so these results are correct.
Only http3/quic work on UDP, and that is not supported.

Also, http2 and http3 actually only work fully on https.

So leave out the `u` from the parameters.

**Comment by Ran-Xing** [comment]
@BlackDex oh ,TKS,I thought it was just the UDP port by default. Let me see. I backed up the data

**Comment by Ran-Xing** [comment]
Sorry , Because I use the Intranet environment, so I signed a certificate, under normal circumstances
The docker run command can be parsed normally, but docker compose up - d just can't. I just applied for a formal SSL certificate, but it doesn't seem to work either

![image](https://user-images.githubusercontent.com/73819467/104846561-1c119900-5916-11eb-8d08-24eaf176f2d4.png)

Docker management software Portainer 1.24

![image](https://user-images.githubusercontent.com/73819467/104846589-419ea280-5916-11eb-9380-194cdd0bcbf0.png)

And I use my own SSL certificate issued by other containers can work normally, but at present only this one is docker-compose up-d can not work normally, docker run can work normally
